components/time/src/provider.rs

```diff
@@ -114,12 +117,106 @@ impl<'a> zerovec::maps::ZeroMapKV<'a> for TimeZone {
 
 /// Storage type for storing UTC offsets as eights of an hour.
 pub type EighthsOfHourOffset = i8;
-/// Storage type for storing `DateTime<Iso>` as minutes since [`EPOCH`].
-pub type MinutesSinceEpoch = i32;
-/// The epoch for [`MinutesSinceEpoch`].
-///
-/// This is 1970-01-01, but this is coincidental to anything UNIX and should be changed to 0 in the future.
-pub const EPOCH: RataDie = RataDie::new(719163);
+/// Storage type for storing `(Date<Iso>, Time)`.
+#[derive(Copy, Clone, PartialEq, PartialOrd, Eq, Ord, Debug)]
+pub struct MinutesSinceEpoch(pub i32);
+
+impl From<(Date<Iso>, Time)> for MinutesSinceEpoch {
+    fn from((d, t): (Date<Iso>, Time)) -> MinutesSinceEpoch {
+        Self(
+            ((Iso::to_fixed(d) - Self::EPOCH) as i32 * 24 + t.hour.number() as i32) * 60
+                + t.minute.number() as i32,
+        )
+    }
+}
+
+impl MinutesSinceEpoch {
+    // This is 1970-01-01, but that is coincidental to anything UNIX and could be changed to 0 in the future.
+    const EPOCH: RataDie = RataDie::new(719163);
+
+    fn to_dt(self) -> (Date<Iso>, Time) {
+        let minute = self.0 % 60;
+        let hour = self.0 / 60 % 24;
+        let days = self.0 / 60 / 24;
+        let date = Iso::from_fixed(Self::EPOCH + days as i64);
+        #[allow(clippy::unwrap_used)] // by %
+        let time = Time::try_new(hour as u8, minute as u8, 0, 0).unwrap();
+        (date, time)
+    }
+}
+
+impl AsULE for MinutesSinceEpoch {
+    type ULE = <i32 as AsULE>::ULE;
+
+    fn from_unaligned(unaligned: Self::ULE) -> Self {
+        Self(i32::from_unaligned(unaligned))
+    }
+
+    fn to_unaligned(self) -> Self::ULE {
+        self.0.to_unaligned()
+    }
+}
+
+impl<'a> ZeroMapKV<'a> for MinutesSinceEpoch {
+    type Container = ZeroVec<'a, Self>;
+    type Slice = ZeroSlice<Self>;
+    type GetType = <Self as AsULE>::ULE;
+    type OwnedType = Self;
+}
+
+#[cfg(feature = "serde")]
+impl serde::Serialize for MinutesSinceEpoch {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        if serializer.is_human_readable() {
+            let (date, time) = self.to_dt();
+            let year = date.year().extended_year;
+            let month = date.month().month_number();
+            let day = date.day_of_month().0;
+            let hour = time.hour.number();
+            let minute = time.minute.number();
+            serializer.serialize_str(&alloc::format!(
+                "{year:04}-{month:02}-{day:02} {hour:02}:{minute:02}"
+            ))
```

Comment:
Suggestion: Use IXDTF syntax.

Reply:
If it wasn't behind a feature I would

Reply:
This is the syntax from CLDR metazones.xml: https://github.com/unicode-org/cldr/blob/2cd040906853ce18c3b87bcf270b3526537c66cc/common/supplemental/metaZones.xml#L23